src/mgmt/srv_pool.c

```diff
@@ -494,13 +494,34 @@ ds_mgmt_create_pool(uuid_t pool_uuid, const char *group, char *tgt_dev,
 	struct mgmt_svc			*svc;
 	d_rank_list_t			*rank_list;
 	uuid_t				*tgt_uuids = NULL;
+	d_rank_list_t			*filtered_targets = NULL;
+	d_rank_list_t			*pg_ranks = NULL;
+	uint32_t			pg_size;
 	int				rc;
 	int				rc_cleanup;
 
 	rc = ds_mgmt_svc_lookup_leader(&svc, NULL /* hint */);
 	if (rc != 0)
 		goto out;
 
+	/* Sanity check targets versus cart's current primary group members.
+	 * If any targets not in PG, flag error before MGMT_TGT_ corpcs fail.
+	 */
+	rc = crt_group_size(NULL, &pg_size);
+	D_ASSERTF(rc == 0, ""DF_RC"\n", DP_RC(rc));
+	pg_ranks = d_rank_list_alloc(pg_size);
+	rc = d_rank_list_dup(&filtered_targets, targets);
+	if (rc) {
+		rc = -DER_NOMEM;
+		D_GOTO(out, rc);
+	}
+	/* Remove any targets not found in pg_ranks */
```

Comment:
This only works as long as we haven't implemented excluding ranks from the system. Once we do have it, the PG may be {0, 2, 5}, where

    pg_size == 3
    pg_ranks = {0, 1, 2}

Reply:
Good point. Although is the current master code in pool_create_prepare() similarly impacted should rank exclude from the system be implemented?

Reply:
I don't fully understand the implications of this... Do we need to address it now or is it something that we can sort out when rank exclude is implemented?

Reply:
We do not have to address it now, of course. I simply would like to note a cleaner approach: this continuous-ranks assumption could be restricted to pool_create_prepare (under the comment "this should consult the system map in the future").